src/Simplex/Chat/Library/Subscriber.hs

```diff
@@ -889,24 +889,45 @@ processAgentMessageConn vr user@User {userId} corrId agentConnId agentMessage =
           -- possible improvement is to choose scope based on event (some events specify scope)
           (gInfo', m', scopeInfo) <- mkGroupChatScope gInfo m
           checkIntegrityCreateItem (CDGroupRcv gInfo' scopeInfo m') msgMeta `catchChatError` \_ -> pure ()
-          forM_ aChatMsgs $ \case
-            Right (ACMsg _ chatMsg) ->
-              processEvent gInfo' m' tags eInfo chatMsg `catchChatError` \e -> eToView e
-            Left e -> do
-              atomically $ modifyTVar' tags ("error" :)
-              logInfo $ "group msg=error " <> eInfo <> " " <> tshow e
-              eToView (ChatError . CEException $ "error parsing chat message: " <> e)
-          forwardMsgs (rights aChatMsgs) `catchChatError` eToView
+          fwdScopesMsgs <- foldM (processAChatMsg gInfo' m' tags eInfo) M.empty aChatMsgs
+          let GroupMember {memberRole = membershipMemRole} = membership
+          when (membershipMemRole >= GRAdmin && not (blockedByAdmin m)) $
+            forM_ (M.assocs fwdScopesMsgs) $ \(groupForwardScope, fwdMsgs) ->
+              forwardMsgs groupForwardScope (L.reverse fwdMsgs) `catchChatError` eToView
           checkSendRcpt $ rights aChatMsgs
         where
           aChatMsgs = parseChatMessages msgBody
           brokerTs = metaBrokerTs msgMeta
-          processEvent :: GroupInfo -> GroupMember -> TVar [Text] -> Text -> MsgEncodingI e => ChatMessage e -> CM ()
+          processAChatMsg ::
+            GroupInfo
+            -> GroupMember
+            -> TVar [Text]
+            -> Text
+            -> Map GroupForwardScope (NonEmpty (ChatMessage 'Json))
```

Comment:
narrowing type to json as we can only forward json (protocol event uses ChatMessage 'Json)